encoder.go

```diff
@@ -16,31 +16,35 @@ import (
 
 	"github.com/DataDog/go-libddwaf/v4/internal/bindings"
 	"github.com/DataDog/go-libddwaf/v4/internal/pin"
-	"github.com/DataDog/go-libddwaf/v4/internal/unsafe"
 	"github.com/DataDog/go-libddwaf/v4/timer"
 	"github.com/DataDog/go-libddwaf/v4/waferrors"
 )
 
-// Encode Go values into wafObjects. Only the subset of Go types representable into wafObjects
+type EncoderConfig struct {
+	// Pinner is used to pin the data referenced by the encoded wafObjects.
+	Pinner pin.Pinner
+	// Timer makes sure the encoder doesn't spend too much time doing its job.
+	Timer timer.Timer
+	// ContainerMaxSize is the maximum number of elements in a container (list, map, struct) that will be encoded.
+	ContainerMaxSize int
+	// StringMaxSize is the maximum length of a string that will be encoded.
+	StringMaxSize int
+	// ObjectMaxDepth is the maximum depth of the object that will be encoded.
+	ObjectMaxDepth int
```

Comment:
Since we're changing, stylistically, I think I kind of would prefer:

- `MaxContainerSize`
- `MaxStringSize`
- `MaxObjectDepth`

So basically the `Max` is always a prefix... it "feels" more coherent to me? Optional, though (the current names aren't too bad either)